hasher-matcher-actioner/hmalib/lambdas/api/datasets_api.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import bottle
+import typing as t
+from dataclasses import dataclass, asdict
+from hmalib.aws_secrets import AWSSecrets
+from threatexchange.api import ThreatExchangeAPI
+from hmalib.common.logging import get_logger
+from .middleware import jsoninator, JSONifiable, DictParseable
+from hmalib.common.config import HMAConfig
+from hmalib.common import config as hmaconfig
+from hmalib.lambdas.fetcher import ThreatExchangeConfig
+from hmalib.common.threatexchange_config import sync_privacy_groups
+
+
+@dataclass
+class Dataset(JSONifiable):
+    privacy_group_id: int
+    privacy_group_name: str
+    fetcher_active: bool
+    write_back: bool
+    in_use: bool
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "Dataset":
+        return cls(
+            d["privacy_group_id"],
+            d["privacy_group_name"],
+            d["fetcher_active"],
+            d["write_back"],
+            d["in_use"],
+        )
+
+
+@dataclass
+class DatasetsResponse(JSONifiable):
+    datasets_response: t.List[Dataset]
+
+    def to_json(self) -> t.Dict:
+        return {
+            "datasets_response": [
+                dataset.to_json() for dataset in self.datasets_response
+            ]
+        }
+
+
+@dataclass
+class SyncDatasetResponse(JSONifiable):
+    response: str
+
+    def to_json(self) -> t.Dict:
+        return asdict(self)
+
+
+@dataclass
+class UpdateDatasetRequest(DictParseable):
+    privacy_group_id: int
+    fetcher_active: bool
+    write_back: bool
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "UpdateDatasetRequest":
+        return cls(
+            d["privacy_group_id"],
+            d["fetcher_active"],
+            d["write_back"],
+        )
+
+
+def get_datasets_api(hma_config_table: str) -> bottle.Bottle:
+    # The documentation below expects prefix to be '/datasets/'
+    datasets_api = bottle.Bottle()
+    HMAConfig.initialize(hma_config_table)
+
+    @datasets_api.get("/", apply=[jsoninator])
+    def datasets() -> DatasetsResponse:
+        """
+        Returns all datasets.
+        """
+        collabs = ThreatExchangeConfig.get_all()
+        return DatasetsResponse(
+            datasets_response=[
+                Dataset(
+                    privacy_group_id=collab.privacy_group_id,
+                    privacy_group_name=collab.privacy_group_name,
+                    fetcher_active=collab.fetcher_active,
+                    write_back=collab.write_back,
+                    in_use=collab.in_use,
+                )
```

Comment:
nit: what I have previously found succinct is to co-locate translations like these. Perhaps a `Dataset.from_collab classmethod` that does these translations?

Then, if anyone is adding an attribute to Dataset, adding the attribute translation is the sensible thing to do.

Reply:
yeah good catch! I added from_collab method for dataset class